feature/platform/ate_tests/telemetry_inventory_test/telemetry_inventory_test.go

```diff
@@ -350,86 +336,77 @@ func TestSwitchChip(t *testing.T) {
 func TestTempSensor(t *testing.T) {
 	dut := ondatra.DUT(t, "dut")
 
-	r := regexp.MustCompile("^TempSensor")
-	sensors := findMatchedComponents(t, dut, r)
-	t.Logf("Found TempSensor list: %v", sensors)
-
+	sensors := components.FindComponentsByType(t, dut, tempsensorType)
+	t.Logf("Found Sensor list: %v", sensors)
 	if len(sensors) == 0 {
 		t.Fatalf("Get TempSensor list for %q: got 0, want > 0", dut.Model())
 	}
 
 	for _, sensor := range sensors {
-		t.Logf("Validate card %s", sensor)
-		component := gnmi.OC().Component(sensor)
-
-		if !gnmi.Lookup(t, dut, component.Id().State()).IsPresent() {
-			// Just log the result using Logf instead of Errorf.
-			t.Logf("component.Id().Lookup(t) for %q: got false, want true", sensor)
-		} else {
-			t.Logf("TempSensor %s Id: %s", sensor, gnmi.Get(t, dut, component.Id().State()))
-		}
+		r := regexp.MustCompile("Temp|TEMP")
+		if r.Match([]byte(sensor)) {
+			t.Logf("Validate card %s", sensor)
+			component := gnmi.OC().Component(sensor)
+
+			if !gnmi.Lookup(t, dut, component.Id().State()).IsPresent() {
```

Comment:
I know this predated this PR, but all these if-statements are doing an unnecessary Get. in the else clause. Better to just do:

if val, ok := gnmi.Lookup(t, dut, component.Id().State()).Val(); ok {
  t.Logf("TempSensor %s Id: %s", sensor, val)
else {
  // Just log the result using Logf instead of Errorf.
  t.Logf("component.Id().Lookup(t) for %q: got false, want true", sensor)
}

To avoid the extra Get call

Reply:
Done